raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+
+        return transaction_hash
+
+    def channel_exists(self, partner: typing.Address) -> bool:
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        log.debug('existing_channel {}'.format(channel_data))
+
+        return channel_data['state'] > CHANNEL_STATE_NONEXISTENT
+
+    def _detail_participant(self, participant: typing.Address, partner: typing.Address):
+        """ Returns a dictionary with the channel participant information. """
+        data = self._call_and_check_result('getChannelParticipantInfo', participant, partner)
+        return {
+            'deposit': data[0],
+            'withdrawn': data[1],
+            'is_closer': data[2],
+            'balance_hash': data[3],
+            'nonce': data[4],
+        }
+
+    def detail_channel(self, partner: typing.Address):
+        """ Returns a dictionary with the channel specific information. """
+        channel_data = self._call_and_check_result('getChannelInfo', self.node_address, partner)
+        return {
+            'channel_identifier': channel_data[0],
+            'settle_block_number': channel_data[1],
+            'state': channel_data[2],
+        }
+
+    def detail_participants(self, partner: typing.Address):
+        """ Returns a dictionary with the participants' channel information. """
+        our_data = self._detail_participant(self.node_address, partner)
+        partner_data = self._detail_participant(partner, self.node_address)
+        return {
+            'our_address': self.node_address,
+            'our_balance': our_data['deposit'],
+            'our_withdrawn': our_data['withdrawn'],
+            'our_is_closer': our_data['is_closer'],
+            'our_balance_hash': our_data['balance_hash'],
+            'our_nonce': our_data['nonce'],
+            'partner_address': partner,
+            'partner_balance': partner_data['deposit'],
+            'partner_withdrawn': partner_data['withdrawn'],
+            'partner_is_closer': partner_data['is_closer'],
+            'partner_balance_hash': partner_data['balance_hash'],
+            'partner_nonce': partner_data['nonce'],
+        }
+
+    def detail(self, partner: typing.Address):
+        """ Returns a dictionary with all the details of the channel and the channel participants.
+        """
+        channel_data = self.detail_channel(partner)
+        participants_data = self.detail_participants(partner)
+
+        return {
+            **channel_data,
+            **participants_data,
+        }
+
+    def locked_amount_by_locksroot(
+            self,
+            participant: typing.Address,
+            partner: typing.Address,
+            locksroot: typing.Locksroot,
+    ) -> int:
+        """ Returns the locked amount for a specific participant's locksroot.
+        """
```

Comment:
Style note for here and all other places where you do it (won't bother putting same comment elsewhere). You don't need to put the closing docstring tags on their own line. Especially if the original line is not over 99 chars.